script/games/gems/DeployGemGame.sol

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Immutable Pty Ltd 2018 - 2023
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.20;
+
+import "forge-std/Test.sol";
+import {GemGame} from "../../../contracts/games/gems/GemGame.sol";
+
+/**
+ * @title IDeployer Interface
+ * @notice This interface defines the contract responsible for deploying and optionally initializing new contracts
+ *  via a specified deployment method.
+ * @dev Credit to axelarnetwork https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/interfaces/IDeployer.sol
+ */
+interface IDeployer {
+    function deploy(bytes memory bytecode, bytes32 salt) external payable returns (address deployedAddress_);
+    function deployAndInit(bytes memory bytecode, bytes32 salt, bytes calldata init)
+        external
+        payable
+        returns (address deployedAddress_);
+    function deployedAddress(bytes calldata bytecode, address sender, bytes32 salt)
+        external
+        view
+        returns (address deployedAddress_);
+}
```

Comment:
@drinkcoffee thoughts on moving this somewhere where it can be reused? Something else we could do is install the contract-deployer repository and potentially import from there 🤔